functable.c

```diff
@@ -16,310 +16,219 @@ static void init_functable(void) {
 
     cpu_check_features();
 
-    // update_hash_stub:
+    // Generic code
```

Comment:
I think that the function assignments throughout this function should be in the same order as listed in the struct in functable.h, starting with adler32.. It seems some of them are out of order. Fixing the order in the if statements would be good too.

Reply:
@nmoinvaz Yes, I was thinking about doing that after #1424, just so there is not too much changes at a time.